sql/src/main/java/org/elasticsearch/action/bulk/BulkShardProcessor.java

```diff
@@ -423,12 +423,16 @@ private void setResultIfDone(int successes) {
     }
 
     private void executeIfNeeded() {
-        if ((closed
+        // we don't want to auto-create indices for every index item,
+        // so execute this only after a certain threshold or if the whole operation is closed/finished.
+        if (((closed && pendingNewIndexRequests.get() > 0)
              || requestsForNewIndices.size() >= createIndicesBulkSize
              || pendingNewIndexRequests.get() >= bulkSize) && failure.get() == null) {
+            // will also execute requests after indices are created
             createPendingIndices();
```

Comment:
I think we should we rename this method to reflect that it's also executing the requests eg createPendingIndicesAndExecuteRequests or something similar (as it's more resilient and expressive than a comment)